publish.js

```diff
@@ -0,0 +1,272 @@
+ *
+ * @param version {String} - the version to set in the package.json file
+ * @param options {Object} - e.g. {"version": "npm version --no-git-tag-version ${version}"}
+ */
+publish.setVersion = function (version, options) {
+    var cmdTemplate = options.version || defaults.version;
+    var cmdStr = es6Template(cmdTemplate, {
+        version: version
+    });
+    publish.execSync(cmdStr);
+};
+
+/**
+ * Calculates the current dev version of the package
+ *
+ * @param options {Object} - e.g. {"rawTimestamp": "git show -s --format=%ct HEAD", "revision": "git rev-parse --verify --short HEAD", "devVersion": "${version}.${timestamp}.${revision}"}
+ * @returns {String} - the current dev version number
+ */
+publish.getDevVersion = function (options) {
+    var rawTimestamp = publish.execSync(options.rawTimestamp || defaults.rawTimestamp);
+    var timestamp = publish.convertToISO8601(rawTimestamp);
+    var revision = publish.execSync(options.revision || defaults.revision);
+    var devVersionTemplate = options.devVersion || defaults.devVersion;
+    var newStr = es6Template(devVersionTemplate, {
+        version: pkg.version,
+        timestamp: timestamp,
+        revision: revision
+    });
+    return newStr;
+};
+
+/**
+ * Publishes the module to npm using the current version number in pacakge.json.
+ * If isTest is specified, it will instead create a tarball in the local directory.
+ *
+ * @param isTest {Boolean} - indicates if this is a test run or not
+ * @param options {Object} - e.g. {"pack": "npm pack", "publish": "npm publish"}
+ */
+publish.pubImpl = function (isTest, options) {
+    if (isTest) {
+        // create a local tarball
+        var packCmd = options.pack || defaults.pack;
+        publish.execSync(packCmd);
+    } else {
+        // publish to npm
+        var pubCmd = options.publish || defaults.publish;
+        publish.execSync(pubCmd);
+    }
+};
+
+/**
+ * Tags the specified version with the specified dist-tag
+ * If it is a test run, the tag command will be output to the console.
+ *
+ * @param isTest {Boolean} - indicates if this is a test run or not
+ * @param version {String} - a string idicating which version to tag
+ * @param tag {String} - the dist-tag to apply
+ * @param options {Object} - e.g. {"distTag": "npm dist-tag add infusion@${version} ${tag}"}
+ */
+publish.tag = function (isTest, version, tag, options) {
+    var cmdTemplate = options.distTag || defaults.distTag;
+    var cmdStr = es6Template(cmdTemplate, {
+        version: version,
+        tag: tag
+    });
+    if (isTest) {
+        console.log("tag command: " + cmdStr);
+    } else {
+        publish.execSync(cmdStr);
+    }
+};
+
+/**
+ * Resets the current workspace.
+ * This will clear out any git tracked changes.
+ *
+ * Used internally to reset version number changes in package.json
+ * @param options {Object} - e.g. {"clean": "git reset HEAD --hard"}
+ */
+publish.clean = function (options) {
```

Comment:
This implementation is quite terrifying and has the potential to lose a lot of user data. I recommend just doing "git checkout package.json"